politeiad/api/v1/v1.go

```diff
@@ -264,3 +289,18 @@ type InventoryReply struct {
 	Vetted   []ProposalRecord `json:"vetted"`   // Last N vetted proposals
 	Branches []ProposalRecord `json:"branches"` // Last N branches (censored, new etc)
 }
+
+// UserErrorReply returns details about an error that occurred while trying to
+// execute a command due to bad input from the client.
+type UserErrorReply struct {
+	ErrorCode    StatusT     `json:"errorcode"`              // Numeric error code
+	ErrorMessage string      `json:"errormessage,omitempty"` // Human readable error string
+	ErrorContext interface{} `json:"errorcontext,omitempty"` // Additional error information
```

Comment:
I really implore you to rethink this.  Having an interface in JSON will lead to tears down the line.  Is there no way to make this generic with a qualifier error code?

Reply:
Changed ErrorContext to []string as we agreed, so the UI doesn't have to be smart about error responses if it doesn't want to.